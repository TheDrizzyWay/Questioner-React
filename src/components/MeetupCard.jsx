import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import dummyImage from '../assets/images/meeting2.jpg';

const styles = {
    card: {
        maxWidth: 300
    },
    media: {
        objectFit: 'cover'
    }
};

const MeetupCard = (props) => {
    const { classes } = props;
    return (
        <Card className={classes.card}>
            <CardActionArea className="meet_card">
                <CardMedia
                    component="img"
                    alt="Contemplative Reptile"
                    className={classes.media}
                    height="170"
                    image={dummyImage}
                />
                <CardContent>
                    <Typography gutterBottom variant="h5" component="h2">
            Sample Meetup
                    </Typography>
                    <Typography component="p">
            LOCATION: Lorem ipsum dolor sit amet, consectetur adipiscing elit,
            sed do eiusmod tempor incididunt
                    </Typography>
                    <Typography component="h3" color="primary" variant="title">
            DATE: 05/05/2019
                    </Typography>
                </CardContent>
            </CardActionArea>
            <CardActions style={{ justifyContent: 'center' }}>
                <Button size="small" color="primary">
          View Details
                </Button>
            </CardActions>
        </Card>
    );
};

MeetupCard.propTypes = {
    classes: PropTypes.object.isRequired
};

export default withStyles(styles)(MeetupCard);
