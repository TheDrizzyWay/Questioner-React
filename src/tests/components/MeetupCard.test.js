import React from 'react';
import { shallow } from 'enzyme';
import MeetupCard from '../../components/MeetupCard';

let wrapper;

describe('<MeetupCard />', () => {
    it('Should render component successfully', () => {
        wrapper = shallow(<MeetupCard />);
        expect(wrapper.length).toEqual(1);
    });

    test('Should maintain existing snapshot', () => {
        wrapper = shallow(<MeetupCard />);
        expect(wrapper).toMatchSnapshot();
    });
});
