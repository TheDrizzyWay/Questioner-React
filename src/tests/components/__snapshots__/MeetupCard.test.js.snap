// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MeetupCard /> Should maintain existing snapshot 1`] = `ShallowWrapper {}`;
